src/tyr/route_serializer_osrm.cc

```diff
@@ -1616,7 +1616,7 @@ summarize_route_legs(const google::protobuf::RepeatedPtrField<DirectionsRoute>&
 
         // k is the number of named segments in the summary. It keeps going
         // up by 1 until route_i's summary is different than the route_j's.
-        size_t k = 1;
+        size_t k = std::min(num_named_segs_needed, num_comparable);
```

Comment:
you got to love these where the fix is one line but the test is like 100 lines :smile: